src/Transport/Topology/TopologyOperator.cs

```diff
@@ -33,14 +33,14 @@ public void Start(TopologySectionInternal topologySection, int maximumConcurrenc
             topology = topologySection;
 
             StartNotifiersFor(topology.Entities);
+            
+            running = true;
 
-            foreach (var operation in pendingStartOperations)
+            Action operation;
+            while (pendingStartOperations.TryTake(out operation))
```

Comment:
The concurrent bag loses the ordering semantics of the list. I think that's OK in this scenario. It can only be subscribe requests and they should not be order dependent.

Reply:
Correct the order doesn't matter